distod/src/main/scala/com/github/codelionx/distod/actors/worker/SplitCandidateValidationBehavior.scala

```diff
@@ -36,46 +37,106 @@ class SplitCandidateValidationBehavior(
 
 
   private val timing = Timing(context.system)
+  private val spans = timing.createSpans
 
   def start(): Behavior[Command] = {
-    context.log.trace("Loading partition errors for all split checks")
+    context.log.trace("Loading partition errors for split checks")
     partitionManager ! LookupError(candidateId, partitionEventMapper)
 
     for (c <- splitCandidates) {
       val fdContext = candidateId - c
       partitionManager ! LookupError(fdContext, partitionEventMapper)
     }
 
-    collectErrors(Map.empty, splitCandidates.size + 1)
+    collectErrors(Map.empty, splitCandidates)
   }
 
-  def collectErrors(errors: Map[CandidateSet, Double], expected: Int): Behavior[Command] =
+  private def collectErrors(errors: Map[CandidateSet, Double], toBeChecked: CandidateSet): Behavior[Command] =
     Behaviors.receiveMessage {
+      case WrappedPartitionEvent(ErrorFound(`candidateId`, value)) =>
+        context.log.trace("Received partition error value: {}, {}", candidateId, value)
+        changeToChecking(value, errors, toBeChecked)
+
       case WrappedPartitionEvent(ErrorFound(key, value)) =>
         context.log.trace("Received partition error value: {}, {}", key, value)
         val newErrorMap = errors + (key -> value)
-        if (newErrorMap.size == expected) {
-          performCheck(newErrorMap)
+        collectErrors(newErrorMap, toBeChecked)
+
+      case m =>
+        stash.stash(m)
+        Behaviors.same
+    }
+
+  private def checking(errorCompare: Double, toBeChecked: CandidateSet, validCandidates: Set[Int]): Behavior[Command] =
+    Behaviors.receiveMessage{
+      case WrappedPartitionEvent(ErrorFound(`candidateId`, _)) =>
+        // ignore, should not happen
+        context.log.error("Received unexpected")
+        Behaviors.same
+
+      case WrappedPartitionEvent(ErrorFound(key, value)) =>
+        context.log.trace("Received partition error value: {}, {}", key, value)
+        spans.begin("Split check")
+        val attributeSet = candidateId diff key
+        val attribute = attributeSet.head
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of head on collections.](https://app.codacy.com/manual/CodeLionX/distod/pullRequest?prid=5011078)